libs/nest-problem-details-filter/src/filters/http-exception.filter.ts

```diff
@@ -17,13 +18,25 @@ export const PROBLEM_CONTENT_TYPE = 'application/problem+json';
 @Catch(HttpException)
 export class HttpExceptionFilter implements ExceptionFilter {
   constructor(
+    @Inject(HttpAdapterHost)
+    private readonly httpAdapterOrHost:
+      | HttpAdapterHost
+      | HttpAdapterHost['httpAdapter'],
```

Comment:
a hacky way to support both the `HttpAdapterHost` provider (from nestjs core) and just the `httpAdapter` property so users of the `new HttpExceptionFilter()` can do:

```ts
import { HttpAdapterHost } from '@nestjs/core' 
// ...
new HttpExceptionFilter(app.get(HttpAdapterHost).httpAdapter)
```

instead of this:

```ts
new HttpExceptionFilter(app.getHttpAdapter)
```